custom/icds_reports/reports/service_delivery_report.py

```diff
@@ -120,24 +120,38 @@ def _location_name():
         return headers
 
     def get_excel_data(self):
+        total_sum_row = [''] * self.config['aggregation_level'] + [0] * (
+            len(self.headers_and_calculation) - self.config['aggregation_level']
+        )
+        total_sum_row[0] = 'Grand Total'
+
         def evaulate_value(row, headers_with_columns):
             row_data = []
 
-            for header in headers_with_columns:
+            for index, header in enumerate(headers_with_columns):
                 if len(header) == 2:
                     if header[1] in (
                         'num_awcs_conducted_cbe',
                         'num_awcs_conducted_vhnd'
                     ) and self.config['aggregation_level'] == 5:
                         row_data.append('Yes' if row[header[1]] == 1 else 'No')
+                        total_sum_row[index] = 'N/A'
                     else:
+                        if header[1] not in self.location_columns:
+                            total_sum_row[index] += row[header[1]]
                         row_data.append(row[header[1]])
                 else:
                     if row[header[2]]:
                         percentage = row[header[1]]/row[header[2]] * 100
                     else:
                         percentage = 0
-                    row_data.append("%.2f" % percentage)
+                    row_data.append("{}%".format("%.2f" % percentage))
+
+                    if total_sum_row[index - 1]:
+                        percentage = total_sum_row[index - 2] / total_sum_row[index - 1] * 100
```

Comment:
Is the idea that the numerator is always in the previous column? It's a little confusing, but I guess if that assumption is clear throughout the report then it seems fine

Reply:
Hey, yeah, this is uniform throughout the report.